sync-core/src/main/java/com/cloudant/sync/datastore/AttachmentManager.java

```diff
@@ -213,13 +240,9 @@ protected Attachment getAttachment(SQLDatabase db, BasicDocumentRevision rev, St
              c = db.rawQuery(SQL_ATTACHMENTS_SELECT,
                      new String[]{attachmentName, String.valueOf(rev.getSequence())});
             if (c.moveToFirst()) {
-                int sequence = c.getInt(0);
-                byte[] key = c.getBlob(2);
-                String type = c.getString(3);
-                int encoding = c.getInt(4);
-                int revpos = c.getInt(7);
+                byte[] key = c.getBlob(c.getColumnIndex("key"));
                 File file = fileFromKey(key);
-                return new SavedAttachment(attachmentName, revpos, sequence, key, type, file, Attachment.Encoding.values()[encoding]);
+                return new SavedAttachment(file, c);
```

Comment:
I'd leave the reading of the values from the database in AttachmentManager, rather than spread database knowledge around. Particularly as you read some values from the cursor here and some in SavedAttachment.